### DISCUSSION TITLE
Watch file changes in module.css and generate module.css.d.ts files

### DISCUSSION THREAD

**Discussion by dmitrijs-balcers** [parcel-bundler/parcel#6488] [Q&A] [answered]
Hello,
I'm struggling with making a transformer (I suppose that is correct type of plugin for this).

What I want to do, I want befriend TS with css modules, and to do that I need to generate `module.css.d.ts` files as siblings to  `module.css` files.
Typically it is being done using `typed-css-modules`.

So I hacked around something like this:
```javascript
const {Transformer} = require('@parcel/plugin');
const DtsCreator = require('typed-css-modules').default;

const DTS_CREATOR = new DtsCreator({camelCase: true});

module.exports = new Transformer({
    async transform({asset}) {
        const dtsContent = await DTS_CREATOR.create(asset.filePath, asset.value);
        await dtsContent.writeFile();
        return [asset];
    },
});
```

And this transformer works like a charm, but only on startup 😕  It does not trigger on file change.
I have tried many different things to make it also get triggered on file change, but nothing worked.
Can anyone point me into right direction please? 😄

**Comment by mischnic** [comment, marked as answer]
It should rerun whenever `asset` changes.

`asset.value` doesn't exist though, you might want `await asset.getCode()` here

**Reply by dmitrijs-balcers** [reply]
Yeah, you are absolutely right.

In the end I even didn't need getCode 👍 

```javascript
const { Transformer } = require('@parcel/plugin');
const DtsCreator = require('typed-css-modules').default;

const creator = new DtsCreator({ camelCase: true });

module.exports = new Transformer({
    async transform({asset}) {
        await (await creator.create(asset.filePath, "", true)).writeFile();
        return [asset];
    },
});

```

**Reply by dmitrijs-balcers** [reply]
Thank you!
